### ISSUE TITLE
Reproducible crash on tab completion

### ISSUE THREAD

**Issue by gwk** [davidhalter/jedi#1369]
I've tried to minimize the example. However I have not tested it on another machine. This is python3.7 on macOS. Let me know if I can do anything else to help diagnose.

m.py:
```
from typing import TypeVar
DateLike = TypeVar('DateLike')
```

s.py:
```
from m import *
import jedi.utils
jedi.utils.setup_readline()
```

Then on the command line, run `python3 -i s.py`.
The script will run and then dump into the REPL. From there, press tab once and it throws this:

```
>>> ERROR:root:REPL Completion error:
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/jedi/cache.py", line 46, in wrapper
    return getattr(self, name)
AttributeError: 'MixedName' object has no attribute '_infer'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/jedi/evaluate/compiled/access.py", line 112, in wrapper
    cache[key]
KeyError: 4359025776

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/jedi/utils.py", line 97, in complete
    completions = interpreter.completions()
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/jedi/api/__init__.py", line 216, in completions
    return completion.completions()
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/jedi/api/completion.py", line 104, in completions
    return sorted(completions, key=lambda x: (x.name.startswith('__'),
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/jedi/api/completion.py", line 44, in filter_names
    k = (new.name, new.complete)  # key
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/jedi/api/classes.py", line 440, in complete
    return self._complete(True)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/jedi/api/classes.py", line 409, in _complete
    if self._name.api_type == 'param' and self._stack is not None:
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/jedi/evaluate/compiled/mixed.py", line 112, in api_type
    return next(iter(self.infer())).api_type
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/jedi/cache.py", line 48, in wrapper
    result = func(self)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/jedi/evaluate/compiled/mixed.py", line 106, in infer
    for c in contexts
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/jedi/common/context.py", line 32, in from_sets
    for set_ in sets:
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/jedi/evaluate/compiled/mixed.py", line 106, in <genexpr>
    for c in contexts
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/jedi/evaluate/compiled/access.py", line 117, in wrapper
    result = func(evaluator, obj, parent_context)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/jedi/evaluate/compiled/mixed.py", line 257, in _create
    assert parent_context.tree_node.get_root_node() == module_node
AssertionError
```

**Comment by davidhalter** [comment]
Can you upgrade Jedi please, this one might have been fixed in the version I just released a few hours ago :)

**Comment by gwk** [comment]
This problem exists on jedi 0.14.1 :(

**Comment by davidhalter** [comment]
Interesting, this seems to happen with Python 3.7, but not 3.5. Thanks for the heads up. Will hopefully be fixed soon.

**Comment by gwk** [comment]
Yes, and I think it's been happening for a long time. I tried Jedi out during the early days of 3.7 and I'm pretty sure I hit the same problem. I only came back to it today!

**Comment by davidhalter** [comment]
This code didn't exist before 3.7, so it must have been another bug :)

**Comment by davidhalter** [comment]
Fixed in ab80646b860d89f0fea2c3aea262cd922c7413d7.
